packages/components/src/Card/Card.tsx

```diff
@@ -53,11 +85,31 @@ export function Card({
     accent && colors[accent],
   );
 
+  const { showCustomCompHeader, showCardHeader } = checkHeaderVisibility(
+    title,
+    header,
+  );
+
   const cardContent = (
     <>
-      {title && (
+      {showCustomCompHeader && header?.customHeader}
+
+      {!showCustomCompHeader && showCardHeader && (
         <div className={styles.header}>
-          <Heading level={3}>{title}</Heading>
+          <div>
+            <Heading level={3}>{header?.title || title}</Heading>
+          </div>
+          {header?.action?.label && (
+            <div className={styles.headerButton}>
```

Comment:
Related to the CSS, I think we should see if we can remove the need for this div + its' associated class.